src/dchmod/dchmod.c

```diff
@@ -708,6 +711,29 @@ static void set_symbolic_bits(const struct perms* p, mfu_filetype type, mode_t*
         }
     }
 
+    /* if assume_all flag is on then calculate the mode 
+     * based on the umask value */
+    if (p->assume_all) {
+        /* lookup current mask and set it back */
+        mode_t old_mask = umask(S_IWGRP | S_IWOTH);
+        umask(old_mask);
+
+        /* only change permission bits that umask allows */
+        *mode &= ~old_mask;
+
+        /* handle mode different if it is a +r,-r, etc */
+        if (p->plus) {
+            /* now OR this with the old mode to not turn off any
+            * bits that were previously turn on */
+            *mode |= old_mode;
+        } else {
+            /* in the case that it was a -r,-w, etc we have to
+             * turn the bits back on that shouldn't be affected
+             * by umask */
+            *mode |= old_mask; 
```

Comment:
Something may be off here.  If I have an existing file with rwxr-xr-x, and then the user does a "-w" with a umask of 077, I think the code here would end up leaving the file with something like r-xrwxrwx when it should be r-xr-xr-x.  Is that right?

I think there could be some similar problem cases with line 728, too.

I think we may need another mask operation in here on old_mode with old_mask somehow.